ext/DFTKPlotsExt.jl

```diff
@@ -108,4 +108,98 @@ function plot_dos(basis, eigenvalues; εF=nothing, unit=u"hartree",
+    # TODO Should also split this up into one stage doing the DOS computation
+    #      and one stage doing the DOS plotting (like now for the bands.)
+
+    n_spin = basis.model.n_spin_components
+    eshift = something(εF, 0.0)
+    εs = range(austrip.(εrange)..., length=n_points)
+
+    # Constant to convert from AU to the desired unit
+    to_unit = ustrip(auconvert(unit, 1.0))
+
+    p = Plots.plot(; kwargs...)
+    spinlabels = spin_components(basis.model)
+    colors = [:blue, :red]
+    Dεs = compute_ldos.(εs, Ref(basis), Ref(eigenvalues), Ref(ψ); smearing, temperature)
+    for σ = 1:n_spin
+        D = [Dσ[σ] for Dσ in Dεs]
+        label = n_spin > 1 ? "LDOS $(spinlabels[σ]) spin" : "LDOS"
+        Plots.plot!(p, (εs .- eshift) .* to_unit, D; label, color=colors[σ])
+    end
+    if !isnothing(εF)
+        Plots.vline!(p, [0.0], label="εF", color=:green, lw=1.5)
+    end
+
+    if isnothing(εF)
+        Plots.xlabel!(p, "eigenvalues  ($(string(unit)))")
+    else
+        Plots.xlabel!(p, "eigenvalues -ε_F  ($(string(unit)))")
+    end
+    p
 end
+plot_ldos(scfres; kwargs...) = plot_ldos(scfres.basis, scfres.eigenvalues, scfres.ψ; scfres.εF, kwargs...)
+
+function plot_pdos(p, i::Integer, l::Integer, iatom::Integer, basis, eigenvalues, ψ; εF=nothing, unit=u"hartree",
+    temperature=basis.model.temperature,
+    smearing=basis.model.smearing,
+    εrange=default_band_εrange(eigenvalues; εF), n_points=1000, kwargs...)
+
+    n_spin = basis.model.n_spin_components
+    eshift = something(εF, 0.0)
+    εs = range(austrip.(εrange)..., length=n_points)
+
+    # Constant to convert from AU to the desired unit
+    to_unit = ustrip(auconvert(unit, 1.0))
+
+    spinlabels = spin_components(basis.model)
+    colbar = [collect(palette(:tab20)), collect(palette(:tab20b))]
+
+    l_orbital = l_orbital_name(l)
+    pdos = compute_pdos(εs, i, l, iatom, basis, eigenvalues, ψ)
+    pname = basis.model.atoms[iatom].psp.pswfc_labels[l+1][i]
+    for σ in 1:n_spin
+        D = pdos[σ]
+        for m = 1:2l+1
+            label = n_spin > 1 ? "$(pname)$(l_orbital[m]) $(spinlabels[σ]) spin" : "$(pname)$(l_orbital[m])"
+            Plots.plot!(p, (εs .- eshift) .* to_unit, D[:, m]; label, color=colbar[σ][l^2+m])
+        end
+    end
+    p
+end
+
+function plot_pdos(i::Integer, l::Integer, iatom::Integer, scfres; kwargs...) 
+    p = plot_dos(scfres; kwargs...)
+    scfres_unfold = DFTK.unfold_bz(scfres)
+    plot_pdos(p, i, l, iatom, scfres_unfold.basis, scfres_unfold.eigenvalues, scfres_unfold.ψ; scfres_unfold.εF, kwargs...)
+end
+
+function plot_pdos(iatom::Integer, scfres; kwargs...)
+    p = plot_dos(scfres; kwargs...)
+    scfres_unfold = DFTK.unfold_bz(scfres)
+    psp = scfres.basis.model.atoms[iatom].psp
+    for l = 0:psp.lmax-1
+        for i = 1:length(psp.r2_pswfcs[l+1])
+            plot_pdos(p, i, l, iatom, scfres_unfold.basis, scfres_unfold.eigenvalues, scfres_unfold.ψ; scfres_unfold.εF, kwargs...)
+        end
+    end
```

Comment:
would you here not want to add the contributions of all l and i channels rather than plotting them separately ? People seem to do that more in practice.